File: matlab/line_prog.m
% 设置参数值
c=[3;3;-4];
a=[-2,3,-1;1,3,1];
b=[-12;10];
aeq=[1,1,1];
beq=6;
% 通用函数
x = linprog(-c, a, b, aeq, beq, zeros(3,1))
value = c'*x

File: Python/line_prog.py
#!/usr/bin/env python
# coding: utf-8

# In[2]:


#Linear programming
from scipy import optimize
import numpy as np


# In[11]:


c = np.array([3, 3, -4])
A = np.array([
    [-2, 3, -1],
    [1, 3, 1]
])
b = ([-12, 10])
Aeq = np.array([[1, 1, 1]])
Beq = np.array([6]) 
res = optimize.linprog(-c, A, b, Aeq, Beq)


# In[8]:


print(res.fun)


# In[9]:


print(res.x)


# In[10]:


print(res)

